File: 백준/Silver/2564. 경비원/경비원.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int w = sc.nextInt(); int h = sc.nextInt();
		int n = sc.nextInt();
		int[] direction = new int[n];
		int[] distance = new int[n];
			for(int i=0; i<n; i++) {
				direction[i] = sc.nextInt();
				distance[i] = sc.nextInt();
			}
		int myDirection = sc.nextInt();
		int myDistance = sc.nextInt();
		
		int sum = 0;
		for(int i=0; i<n; i++) {
			if(myDirection == direction[i]) {
				sum+= Math.abs(myDistance-distance[i]);
			}else {
				if(myDirection ==1 ) {
					if(direction[i] == 2) {
						sum += h+Math.min(distance[i]+myDistance, Math.abs(w-distance[i]+w-myDistance));
						
					}else if (direction[i]==3) {
						sum+= distance[i]+myDistance;
					}else {//4
						sum+=distance[i]+w-myDistance;
					}
					
				}else if(myDirection == 2) {
					if(direction[i] == 1) {
						sum += h+Math.min(distance[i]+myDistance, w-distance[i]+w-myDistance);
					}else if (direction[i]==3) {
						sum+= h-distance[i]+myDistance;
					}else {
						sum+=h-distance[i]+w-myDistance;
					}
				}else if(myDirection ==3) {
					if(direction[i] == 4) {
						sum += w+Math.min(distance[i]+myDistance, h-distance[i]+h-myDistance);
						
					}else if (direction[i]==1) {
						sum+= distance[i]+myDistance;
					}else {//2
						sum+=w-distance[i]+myDistance;
					}
				}else { //4
					if(direction[i] == 3) {
						sum += w+Math.min(distance[i]+myDistance, h-distance[i]+h-myDistance);
						
					}else if (direction[i]==1) {
						sum+= w-distance[i]+myDistance;
					}else {//2
						sum+=w-distance[i]+myDistance;
					}
				}
			}
		
		}
		System.out.println(sum);
	}

}

File: 백준/Silver/2564. 경비원/경비원.py
w, h = map(int, input().split())

n = int(input())
answer = 0

shops = []
for i in range(n):
    shops.append(list(map(int, input().split())))

direction, dist = map(int, input().split())

for a in shops:
     if a[0] == direction:
         answer += abs(a[1] - dist)
     elif direction == 1: #북쪽
         if a[0] == 2: #남
             answer += h + min(dist + a[1], 2 * w-dist-a[1])
         elif a[0] == 3: #서
             answer += a[1] + dist
         else:
             answer += w-dist + a[1]
     elif direction == 2:
         if a[0] == 1: #북
             answer += h + min(dist + a[1], 2*w-dist-a[1])
         elif a[0] == 3: #서
             answer += h-a[1] + dist
         else:#동
             answer += w-dist + h-a[1]
     elif direction == 3:#서
         if a[0] == 1: #북
             answer += dist + a[1]
         elif a[0] == 2: #남
             answer += h-dist+a[1]
         else:#동
             answer += w+min(dist+a[1], 2*h-dist-a[1])
     elif direction ==4: #동
         if a[0] == 1: #북
             answer += dist+w-a[1]
         elif a[0] == 2: #남
             answer += h-dist+w-a[1]
         else: #서
             answer += w + min(a[1]+dist, 2*h-dist-a[1])

print(answer)


